### ISSUE TITLE
eval=FALSE does not work in combination with cleanlog=FALSE

### ISSUE THREAD

**Issue by jules-bec** [Hemken/Statamarkdown#33]
Dear Mr. Hemken,

I have a problem with displaying code. I want to display certain Stata commands without showing the respective result. It is possible to use the chunk options "eval=FALSE" or "results="hide"" to do so, however these chunk options don't seem to be compatible with the chunk option "cleanlog=FALSE", as it does not show anything when used together, not the result and not the command. As I would like to display Stata code in the typical black and white format, this is somewhat bothersome. Is there a way to suppress the output and merely show the Stata commands while still using "cleanlog=FALSE" ? I have attached a pdf to exemplify my problem and would be very thankful for your help!

Kind regards, 
Julien

[Stata-Block-Test.pdf](https://github.com/Hemken/Statamarkdown/files/10475523/Stata-Block-Test.pdf)

**Comment by Hemken** [comment]
You might try using the “verbatim “ engine, rather than the Stata engine.

With the eval=FALSE option, Stata does not run and no log is produced.  With results=‘hide’, Stata runs but no output is shown.  With cleanlog=FALSE, Stata’s command echo in the log is left intact, as would be normal when working in Stata.  The echo=FALSE is an independent option that suppresses the independent code block shown in a document.
________________________________
From: jules-bec ***@***.***>
Sent: Sunday, January 22, 2023 3:20:04 PM
To: Hemken/Statamarkdown ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Hemken/Statamarkdown] eval=FALSE does not work in combination with cleanlog=FALSE (Issue #33)


Dear Mr. Hemken,

I have a problem with displaying code. I want to display certain Stata commands without showing the respective result. It is possible to use the chunk options "eval=FALSE" or "results="hide"" to do so, however these chunk options don't seem to be compatible with the chunk option "cleanlog=FALSE", as it does not show anything when used together, not the result and not the command. As I would like to display Stata code in the typical black and white format, this is somewhat bothersome. Is there a way to suppress the output and merely show the Stata commands while still using "cleanlog=FALSE" ? I have attached a pdf to exemplify my problem and would be very thankful for your help!

Kind regards,
Julien

Stata-Block-Test.pdf<https://github.com/Hemken/Statamarkdown/files/10475523/Stata-Block-Test.pdf>

**Comment by remlapmot** [comment]
It's also worth pointing out that you can simply use a code chunk without fences, i.e., a code chunk such as

````
```stata
display "Here's some Stata code that won't be executed"
```
````

You only need the verbatim approach if you want to show the backticks and fences, i.e.,

`````
````{verbatim}
```{stata}
display "Some more Stata code that won't be executed"
```
````
`````

**Comment by remlapmot** [comment]
And in a Quarto document as an equivalent to the verbatim syntax you can use the double `{{}}` approach

````
```{{stata}}
display "Some Stata code that won't be executed"
```
````

**Comment by Hemken** [comment]
Thanks for pointing that out.

And glad to see someone is experimenting with how everything interacts with Quarto!

--
Doug Hemken
Statistical consultant
4226I Social Science Bldg.

***@***.***
608-262-4327

To make a consulting appointment send me an email, or use the on-line scheduler:
https://doodle.com/mm/doughemken/book-a-time

**Comment by remlapmot** [comment]
Absolutely, I find that Statamarkdown works similarly great with Quarto's knitr engine.

**Comment by remlapmot** [comment]
But just to clarify to you that for all 3 of those syntaxes I gave Statamarkdown is not actually required - because they are just verbatim syntaxes generating markdown (well the first is already plain markdown).

In those syntaxes you could actually write almost anything you wanted instead of `stata`. In those syntaxes when we write `stata` the generated markdown gets Stata syntax highlighting at least on GitHub because Stata is a language included in Linguist. The entry for Stata is [here](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/languages.yml#L6501).

Just to show this here are example Rmd and qmd files.

Example Rmd file
`````rmarkdown
---
output: html_document
---

```stata
display "Here's some Stata code that won't be executed"
```

````{verbatim}
```{stata}
display "Some more Stata code that won't be executed"
```
````
`````

Example qmd file
`````rmarkdown
---
format:
  html:
    embed-resources: true
engine: knitr
---

```stata
display "Here's some Stata code that won't be executed"
```

````{verbatim}
```{stata}
display "Some more Stata code that won't be executed"
```
````

```{{stata}}
display "Some Stata code that won't be executed"
```
`````
